fbpic/particles/elementary_process/ionization/ionizer.py

```diff
@@ -267,18 +267,15 @@ def handle_ionization( self, ion ):
                 ion.ux, ion.uy, ion.uz, ion.Ex, ion.Ey, ion.Ez,
                 ion.Bx, ion.By, ion.Bz, ion.w, self.w_times_level )
 
-        # Count the total number of new electrons (operation always performed
-        # on the CPU, as this is typically difficult on the GPU)
-        if use_cuda:
-            n_ionized = n_ionized.get()
-        cumulative_n_ionized = perform_cumsum_2d( n_ionized )
+        # Count the total number of new electrons 
+        cumulative_n_ionized = perform_cumsum_2d( n_ionized, use_cuda )
         # If no new particle was created, skip the rest of this function
-        if np.all( cumulative_n_ionized[:,-1] == 0 ):
-            return
-        # Copy the cumulated number of electrons back on GPU
-        # (Keep a copy on the CPU)
         if use_cuda:
-            d_cumulative_n_ionized = cupy.asarray( cumulative_n_ionized )
+            if cupy.all( cumulative_n_ionized[:,-1] == 0 ):
```

Comment:
The first memcpy stems from here - `cupy.all` runs on the GPU (as it has to), the resulting `bool` value thus has to be copied to the CPU in order for Python to do the `if` check.